fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/FromItem.java

```diff
@@ -1,22 +1,84 @@
 /*
- * (C) Copyright IBM Corp. 2019
+ * (C) Copyright IBM Corp. 2019,2021
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 package com.ibm.fhir.database.utils.query;
 
+import com.ibm.fhir.database.utils.query.expression.StatementRenderer;
+
 /**
- * An entry in the FROM clause
+ * An element in the FROM clause
  */
 public class FromItem {
+
+    // The source we are joining to
+    private final RowSource rowSource;
+
+    // Optional alias...but recommended to avoid nasty surprises
     private final Alias alias;
 
-    protected FromItem(Alias alias) {
+    /**
+     * @param rowSource
+     * @param alias
+     */
+    protected FromItem(RowSource rowSource, Alias alias) {
+        this.rowSource = rowSource;
         this.alias = alias;
     }
 
+    protected FromItem(RowSource rowSource) {
+        this.rowSource = rowSource;
+        this.alias = null;
+    }
+
+    /**
+     * Get the alias associated with this item in the from clause.
+     * @return the alias for this item. Can be null
+     */
     public Alias getAlias() {
+        // can be null
         return this.alias;
     }
+
+    /**
+     * Is this an ANSI-style join?
+     * @return
+     */
+    public boolean isAnsiJoin() {
+        return false;
+    }
+
+    @Override
+    public String toString() {
+        return toPrettyString(false);
+    }
+
+    public String toPrettyString(boolean pretty) {
+        StringBuilder result = new StringBuilder();
+        result.append(rowSource.toPrettyString(pretty));
+
+        // the alias can be null - it's not always needed, but recommended
+        Alias aliasName = getAlias();
+        if (aliasName != null) {
+            result.append(" AS ").append(aliasName.toString());
+        }
+
+        return result.toString();
+    }
+
+    public <T> T render(StatementRenderer<T> renderer) {
```

Comment:
feels like this should be extracted to an interface?

Reply:
Eventually I want to align the whole Select statement and its components with the ExpNode stuff so we have a single visitor pattern. Not worth changing anything now until we get that done.

Reply:
Seems  like a good  backlog issue as well

Reply:
#2321